### PR TITLE
Refactor Mistral to use PostgreSQL instead of MySQL

### PR DESCRIPTION
**Pull request by m4dcoder**

### CODE REVIEW

**Comment by lakshmi-kannan** [comment]
This also needs a corresponding change in workroom.

File: tools/st2_deploy.sh [hunk 1]
```diff
@@ -97,6 +99,22 @@ else
     MISTRAL_STABLE_BRANCH="st2-0.5.1"
 fi
 
+update_hosts_file()
+{
+  if [ -z "`cat /etc/hosts | grep -e '127.0.0.1\slocalhost'`" ]; then
```
Comment (lakshmi-kannan) [inline tools/st2_deploy.sh]: Just noting in workroom, we use 127.0.1.1.

**Comment by manasdk** [comment]
Are we sure about this change? I don't think we should be changing anyones /etc/hosts without notifying them - if I have some terrible reason for this configuration I would hate it if some deploy script went in an changed things!

File: tools/st2_deploy.sh (hunk 1 above)
Comment (m4dcoder) [inline tools/st2_deploy.sh]: > Are we sure about this change? I don't think we should be changing anyones /etc/hosts without notifying them - if I have some terrible reason for this configuration I would hate it if some deploy script went in an changed things!

I'm open to suggestions. I don't know how else to fix this on vagrant w/ chef/fedora-20.

Comment (lakshmi-kannan) [inline tools/st2_deploy.sh]: IIUC, this basically sets localhost to 127.0.0.1. Why can't you edit to configuration for the service (postgres in this) on the box and replace localhost with whatever IP we want?

Comment (manasdk) [inline tools/st2_deploy.sh]: Yeah that seems like a better approach.

Comment (m4dcoder) [inline tools/st2_deploy.sh]: st2api and st2auth are the one causing problem. actually this doesn't have anything to do with the postgresql change.  i can happily remove this and let the f20 fail for now.

Comment (manasdk) [inline tools/st2_deploy.sh]: so st2api and st2auth have always failed on fedora and no one has complained?

Comment (m4dcoder) [inline tools/st2_deploy.sh]: Looks like it.

Comment (DoriftoShoes) [inline tools/st2_deploy.sh]: It wasn't always this way.  I don't know when it changed but in the beginning it wasn't happening.

**Comment by DoriftoShoes** [comment]
st2_deploy.sh is designed for standalone deployments for testing/POC purposes.  I don't have a problem with us ensuring that the localhost entry we are looking for exists on these boxes.  Now if we were doing this as part of a package install it would be a different story...but this all in one installer isn't designed to be dropped in to a production environment.

File: tools/st2_deploy.sh (hunk 1 above)
Comment (m4dcoder) [inline tools/st2_deploy.sh]: Anyway, I found a fix. The problem is in the st2client default base URL.  I'm going to set it to http://0.0.0.0 (currently it's http://localhost).  That fixed the problem.

Comment (m4dcoder) [inline tools/st2_deploy.sh]: The fix is done separately @ https://github.com/StackStorm/st2/pull/1609.

**Pull request merged**